### PR TITLE
First attempt at a FAQ

### PR DESCRIPTION
**Pull request by choldgraf**
Inspired by @yuvipanda , I put together a short FAQ for the user-facing Binder docs, and rearranged a bit of `index` since it already had some FAQ-ey content in it. Lemme know what y'all think! I imagine we'll need to iterate a bit on the wording / branding / etc but this is a first step :-)

Link to built docs:

http://predictablynoisy.com/binder/

### CODE REVIEW

**Comment by Carreau** [comment]
There is a small conflict. I can resolve it online.

@choldgraf do you want to work on this or should we got with is as is ?

File: doc/reproducibility.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Tips for ensuring reproducibility with Binder
```
Comment (willingc) [inline doc/reproducibility.rst]: Tips for reproducibility

File: doc/reproducibility.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+=============================================
+
+Binder will create a new Docker image the first time is run with a repository.
+From then on, a new image will only be created if the commit hash changes
+(if you're linking Binder to a branch and not a specific commit hash). Here
+are some tips to ensure reproducibility of your Binder links even if you must
+re-build your repository image:
+
+Pin your dependencies
```
Comment (willingc) [inline doc/reproducibility.rst]: ~~your~~

**Comment by choldgraf** [comment]
ok I think I addressed all comments etc, LMK!

**Comment by choldgraf** [comment]
ping! maybe @betatim wants to take a look (or press the big green button :-D )

**Pull request merged**